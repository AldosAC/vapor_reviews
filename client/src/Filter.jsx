import React from 'react';
import styled from 'styled-components';

class Filter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showMenu: false
    };
  }

  showMenu(e) {
    e.preventDefault();
    this.setState({
      showMenu: true
    });
  }

  hideMenu(e) {
    e.preventDefault();
    this.setState({
      showMenu: false
    });
  }

  render() {
    const onMouseLeave = {
      borderRadius: '0px',
      display: 'inline-block',
      color: '#4582a5',
      padding: '10px',
      paddingRight: '10px',
      background: '#1f2e42',
      fontSize: '10px'
    };

    const onMouseEnter = {
      borderRadius: '0px',
      display: 'inline-block',
      color: '#4582a5',
      padding: '10px',
      paddingRight: '10px',
      background: '#c6d4df',
      fontSize: '10px'
    };

    const setFilters = this.props.setFilters;

    return (
      <FilterDropdown onMouseEnter={this.showMenu.bind(this)} onMouseLeave={this.hideMenu.bind(this)}>
        <FilterButton style={this.state.showMenu ? onMouseEnter : onMouseLeave }>{this.props.filter.displayName.toUpperCase()}</FilterButton>
        {
          this.state.showMenu ? 
            (
              <FilterMenu>
                <FilterOption>
                  <RadioButton onClick={(e) => setFilters(e, this.props.filter.id, {})} name={this.props.filter.id} type="radio" id='all'/>
                  <RadioLabel onClick={(e) => setFilters(e, this.props.filter.id, {})} htmlFor='all'>All</RadioLabel>
                  <OptionCount>({this.props.filter.options.reduce((total, current) => (total + current.count), 0)})</OptionCount>
                </FilterOption>
                {
                  this.props.filter.options.map(option => {
                    let optionId = option.id;
                    let optionName = option.displayName;
                    let count = option.count;

                    return (
                      <FilterOption key={optionId}>
                        <RadioButton onClick={(e) => setFilters(e, this.props.filter.id, optionId)} name={this.props.filter.id} type="radio" id={optionId}/>
                        <RadioLabel onClick={(e) => setFilters(e, this.props.filter.id, optionId)} htmlFor={optionId}>{optionName}</RadioLabel>
                        <OptionCount>({count})</OptionCount>
                      </FilterOption>
                    );
                  })
                }
              </FilterMenu>
            )
            : null
        }
      </FilterDropdown>
    );
  }
}

const FilterDropdown = styled.div`
  display: inline-block;
  position: relative;
`;

const FilterButton = styled.div`

`;

const FilterMenu = styled.ul`
  margin: 0;
  padding: 5px 0px;
  top: 30px;
  left: 0px;
  width: 200px;
  background: #c6d4df;
  position: absolute;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  z-index: 1;
`;

const FilterOption = styled.li`
  padding: 3px 5px;
  background: #c6d4df;
  color: #556772;
  font-size: 12px;
  font-weight: normal;
`;

const RadioButton = styled.input`
`;

const RadioLabel = styled.label`
`;

const OptionCount = styled.div`
  display: inline;
  color: #7193a6;
  padding-left: 3px;
`;

export default Filter;