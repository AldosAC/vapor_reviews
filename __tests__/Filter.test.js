import React from 'react';
import { shallow } from 'enzyme';
import Filter from '../client/src/Filter.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let props;
let mockCallback;

const createProps = (mockCallback) => ({
  setFilters: mockCallback,
  filter: {
    id: 'test_id',
    displayName: 'display_name'
  }
});

beforeEach(() => {
  jest.resetModules();
  mockCallback = jest.fn();
  props = createProps(mockCallback);
  wrapper = shallow(<Filter {...props} /> );
});

describe('<Filter /> rendering', () => {
  it('should render elements', () => {
    expect(wrapper).toMatchSnapshot();
  });
});