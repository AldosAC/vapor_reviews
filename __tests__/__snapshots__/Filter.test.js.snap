// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> rendering should render elements 1`] = `
<styled.div
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <styled.div
    hover={false}
  >
    DISPLAY_NAME
  </styled.div>
</styled.div>
`;
